### PR TITLE
WIP: Variable view improvements and color-coding

### PR DESCRIPTION
**Pull request by fRedelaar**
**Variable view: frontend improvements and color-coding**. 

**Description**

For issue #1329 , I'm working on a view that displays variables and their values. In this PR I've added many  improvements. Here are the most important ones:

- Created a togglable button in the output window. Clicking on this will expand/collapse the variable view. 
- Made variable view visually look like the earlier shown designs at #1329 
- Made adjustments so this newly made toggle won't disappear when clicking on tabs/adventures, or when running a program. (Both flush the output window.)
- Added drop-down symbol on button
- removed int_svr variables in variable view
- Made the title "Variables" translatable. Added it in NL and ENG. 
- **Added color-coding for variable types.** At the moment all variable types (string, number, list and boolean) are shown as white text. This is customizable however! Take a look at function `special_style_for_variable` at `app.ts`. Just adjust the hex colors and voilà! 

**How to test**
set the feature flag variable "`variable_view`" to "`true`" at app.ts. Then go to any level that uses variables. Level 2 for example. Click on the blue "variables" button such as the screenshot below and the list will expand/collapse.

**Screenshot**

https://user-images.githubusercontent.com/33595265/158625032-1a6af9db-0949-4b97-a6b2-d23bdf8e0a74.mov

### CODE REVIEW

**Comment by Felienne** [comment]
Hi @fRedelaar! If a PR is WIP (i.e. not ready to be reviewed/merged) you can set it to draft so we really can't merge it.

You can do that on creation but also now still, at the top right:

image

**Comment by fRedelaar** [comment]
Hi @Felienne , I think it is ready to be reviewed, but I wrote WIP because the project isn't finished 100% . I thought it would be good to merge in chunks :) But if you want I can set it to draft!

**Comment by Felienne** [comment]
@TiBiBa This can be merged but.... it adds texts so I am afraid to add it before you finish the Babel thing. Can you merge this when you are ready
?

File: templates/incl-editor-and-output.html
Comment (TiBiBa) [inline templates/incl-editor-and-output.html:91]: It would be nice if we solved this with a bit more "css friendly" styling. For example with something like this:

`ltr:right-0 rtl:left-0 absolute z-10 mx-2 mt-2`

Instead of styling the button we should be styling the dropdown-menu itself, containing the button.

File: templates/incl-editor-and-output.html
Comment (TiBiBa) [inline templates/incl-editor-and-output.html:96]: This would also be nice to be styled a bit "cleaner". The code will be hard to maintain if we ever want to change the UI of our code page.

**Review by TiBiBa — APPROVED** [review]
Looks good to me! I only left some comments concerning the css styling. If it works, it works. But for our future self it would be great to try and solve it with a bit of cleaner CSS code.

**Review by Felienne — CHANGES_REQUESTED** [review]
Setting this to request Changes so @fRedelaar can fix the small CSS issues.

File: templates/incl-editor-and-output.html
Comment (fRedelaar) [reply templates/incl-editor-and-output.html]: After some puzzling, I've replaced most of the CSS inline code with tailwind classes :) I hope its better now

**Comment by fRedelaar** [comment]
Removed translation of variable for now. It will be added back in later PR. (Once I understand Babel better..)

**Comment by TiBiBa** [comment]
> I've replaced most of the inline CSS with tailwind classes.
> 
> P.S. Also removed the translation of the word "variable" for now. It will be added back in later PR. (Once I understand Babel better..)

Awesome! Will take a look!

**Comment by fRedelaar** [comment]
@TiBiBa I've tested it, and it looks great!! It still works like it used to do. Thank you for the tweaks :) I'll take a look at what you've done to learn from it

**Comment by Felienne** [comment]
Ehm @TiBiBa to the rescue for the babel stuff here?

**Comment by TiBiBa** [comment]
> Ehm @TiBiBa to the rescue for the babel stuff here?

Fixed!

**Comment by Felienne** [comment]
Guess we again have a conflict here 😢 ?

**Comment by TiBiBa** [comment]
Will take a look!

**Comment by TiBiBa** [comment]
Fixed!

**Review by TiBiBa — APPROVED** [review]

**Review by Felienne — APPROVED** [review]

**Pull request merged**
